cli50

```diff
@@ -31,7 +32,40 @@ def main():
     if not args["fast"]:
         subprocess.call(["docker", "pull", "cs50/cli"])
 
-    # mount directory in CS50 CLI
+    # check for running containers
+    stdout = subprocess.check_output([
+        "docker", "ps",
+            "--all",
+            "--filter",
+            "volume={}".format(directory),
+            "--format", "{{.ID}}\t{{.Image}}\t{{.RunningFor}}"
+            ]).decode("utf-8")
+    containers = []
+    for line in stdout.rstrip().splitlines():
+        ID, Image, RunningFor = line.split("\t")
+        if Image == "cs50/cli":
```

Comment:
Wanted to do that, but seems `ancestor` is literally that, any ancestor, in which case if a user were running another image that descends from `cs50/cli` (e.g., `cs50/server`), that would be mistaken for CS50 CLI too if we used `ancestor`.